lib/parsing.js

```diff
@@ -25,23 +25,73 @@
  * @property {AstNode[]} [params]
  */
 
-const {SyntaxError: JSDocTypeSyntaxError, parse} = require('../peg_lib/jsdoctype.js');
+const {
+  parse, SyntaxError: JSDocTypeSyntaxError,
+} = require('../peg_lib/jsdoctype-permissive.js');
+
+const {
+  parse: parseJsdoc, SyntaxError: JSDocSyntaxError,
+} = require('../peg_lib/jsdoctype-jsdoc.js');
+const {
+  parse: parseClosure, SyntaxError: ClosureSyntaxError,
+} = require('../peg_lib/jsdoctype-closure.js');
+const {
+  parse: parseTypeScript, SyntaxError: TypeScriptSyntaxError,
+} = require('../peg_lib/jsdoctype-typescript.js');
 
 module.exports = {
 
-  /** * A class for JSDoc type expression syntax errors.
+  /**
+   * A class for JSDoc-like type expression syntax errors.
    * @constructor
    * @extends {Error}
    */
   JSDocTypeSyntaxError,
 
+  /**
+   * A class for JSDoc-specific type expression syntax errors.
+   * @constructor
+   * @extends {Error}
+   */
+  JSDocSyntaxError,
+
+  /**
+   * A class for Closure type expression syntax errors.
+   * @constructor
+   * @extends {Error}
+   */
+  ClosureSyntaxError,
+
+  /**
+   * A class for TypeScript type expression syntax errors.
+   * @constructor
+   * @extends {Error}
+   */
+  TypeScriptSyntaxError,
+
+  /**
+   * @typedef {object} ParseOptions
+   * @property {"permissive"|"jsdoc"|"closure"|"typescript"} [opts.mode='permissive']
+   */
 
   /**
    * Parse the specified type expression string.
    * @param {string} typeExprStr Type expression string.
+   * @param {ParseOptions} opts
    * @return {AstNode} AST.
    */
-  parse (typeExprStr) {
-    return parse(typeExprStr);
+  parse (typeExprStr, {mode = 'permissive'} = {}) {
+    switch (mode) {
+      case 'jsdoc':
+        return parseJsdoc(typeExprStr);
+      case 'closure':
+        return parseClosure(typeExprStr);
+      case 'typescript':
+        return parseTypeScript(typeExprStr);
+      case 'permissive':
+        return parse(typeExprStr);
+      default:
+        throw new TypeError('Unrecognized mode!');
```

Comment:
It&nbsp;might be&nbsp;a&nbsp;good&nbsp;idea to&nbsp;log what&nbsp;was&nbsp;passed:
```suggestion
        throw new TypeError('Unrecognized mode: "' + mode + '"');
```

Even better would probably be&nbsp;to&nbsp;use&nbsp;[`util.inspect(…)`](https://nodejs.org/dist/latest-v12.x/docs/api/util.html#util_util_inspect_object_showhidden_depth_colors).

Reply:
Thanks for the suggestions! I've applied...

I'm open to `util.inspect`, but I'm not sure how you'd like it used here.

Reply:
And I've been a bit busy these days with other items to complete the PR, needing as it does some extra research.

Reply:
Using `util.inspect` means:
```js
throw new TypeError('Unrecognized mode: ' + util.inspect(mode));
```

But `util` first has to be imported:
```js
const util = require('util');
```

Reply:
Ok, thanks, I wasn't sure if you meant logging the object and then throwing with the current message.